device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -184,6 +184,33 @@ public void sendEventAsync(Message message, IotHubEventCallback callback, Object
         deviceIO.sendEventAsync(message, callback, callbackContext, this.config.getDeviceId());
     }
 
+    /**
+     * ASynchronously sends a batch of messages to the IoT Hub
+     * HTTPS messages will be sent in a single batch and MQTT and AMQP messages will be sent individually.
```

Comment:
MQTT and AMQP messages will be sent individually - protocol library restriction?

Reply:
yes, service does not support MQTT batch messages, and our current AMQP library (proton) doesn't have built in batch message capabilities so we have to implement our own. That will be done separately.